modules/sdk-coin-sui/src/lib/transaction.ts

```diff
@@ -176,7 +177,7 @@ export abstract class Transaction<T> extends BaseTransaction {
     } else if (txDetails.hasOwnProperty('PayAllSui')) {
       type = SuiTransactionType.PayAllSui;
     } else if (txDetails.hasOwnProperty('Call')) {
-      type = SuiTransactionType.AddDelegation;
+      type = utils.getSuiTransactionType(_.get(txDetails, 'Call.function'));
```

Comment:
Lodash performs much slower than native typescript, so it's better to avoid using lodash in typescript files.
```suggestion
      const moveCallTxDetail = txDetails as MoveCallTxDetails;
      type = utils.getSuiTransactionType(moveCallTxDetail.Call.function);
```

Reply:
ok fixing